client/src/components/BuddyMatcher.jsx

```diff
@@ -28,7 +32,10 @@ const BuddyMatcher = () => {
     const startNewSearch = () => {
         startSearch();
         setLoadingText(defaultLoadingText);
-        socket.emit('join', { loginId: authState.loginId, email: authState.email });
+        socket.volatile.emit('join', {
+            loginId: authState.loginId,
+            email: authState.email,
+        });
```

Comment:
what's the reason for this change here?

Reply:
To avoid sending that event multiple times to the server once the connection is back

Reply:
sounds good, i noticed it's only this emit that uses it and not the other login emit, why?

Reply:
Which login?

Reply:
Search for "socket. Emit(" login")" on vs code, all other results in the client is what I mean

Reply:
I can't find any results for `socket.emit('login')`, the closest I can find is `socket.emit('logout')` is there a particular file or line I can check?

Reply:
Oh, I've just realised there's already a `join` event being emitted once the socket connects. I just need to connect. Plus I need make all the `join` events volatile.

I guess you were referring to the second `join` event at line 128-131 (this same file)?

Reply:
Yes and every instance of a join socket event, thought there was a specific reason why you didn't change that too